tests/functional/u/undefined/undefined_all_variable_edge_case.py

```diff
@@ -0,0 +1,5 @@
+"""Edge case: __all__ exists in module's locals, but cannot be inferred.
+
+Other tests for undefined-all-variable in tests/functional/n/names_in__all__.py"""
+
+__all__ += []  # [undefined-variable]
```

Comment:
I'm not well-versed in using `__all__`. Is it the case that it doesn't exist until it's defined explicitly? (As opposed to being an always-defined global)

Reply:
I probably shouldn't have used the word "exists".

When __all__ isn't defined, you lose the ability to do:
```python
>>> from pylint import *
>>> version
'3.0.0b1'
```

> Is it the case that it doesn't exist until it's defined explicitly? (As opposed to being an always-defined global)

Indeed!
```python
>>> import astroid
>>> astroid.__all__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'astroid' has no attribute '__all__'. Did you mean: '__file__'?
```

Reply:
Versus
```python
>>> pylint.__all__
['__version__', 'version', 'modify_sys_path', 'run_pylint', 'run_symilar', 'run_pyreverse']
```